### PR TITLE
DOC: Explain pivot vs. pivot_table (#6950)

### LINKED ISSUE #6950: ENH/CLN: Make df.pivot to the mapping of pivot_table
Related to #6913.

`DataFrame.pivot` function uses separate logic and has less keyword options than `pandas.tools.pivot_table`. How about making `DataFrame.pivot` to call `pivot_table` internally to support same keywords as `pivot_table`?

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] closes #6950

### CODE REVIEW

File: doc/source/reshaping.rst [hunk 1]
```diff
@@ -90,6 +90,11 @@ You of course can then select subsets from the pivoted DataFrame:
 Note that this returns a view on the underlying data in the case where the data
 are homogeneously-typed.
 
+While ``pivot`` provides general purpose pivoting of DataFrames with various
+data types (strings, numerics, etc.), Pandas also provides the ``pivot_table``
+function for pivoting with aggregation of numeric data (see the
```
Comment (jreback) [inline doc/source/reshaping.rst]: i would love this down to the first paragraph in the pivot section

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tools/pivot.py [hunk 2]
```diff
@@ -75,6 +75,11 @@ def pivot_table(data, values=None, index=None, columns=None, aggfunc='mean',
     Returns
     -------
     table : DataFrame
+
+    See also
+    --------
+    DataFrame.pivot : pivot without aggregation that can handle
+        non-numeric data
```
Comment (mroeschke) [inline pandas/tools/pivot.py]: The `pivot` docstring already references `pivot_table` ([here](https://github.com/pandas-dev/pandas/blob/master/pandas/core/frame.py#L3908))

File: doc/source/reshaping.rst (hunk 1 above)
Comment (mroeschke) [inline doc/source/reshaping.rst]: Sounds good @jreback. I'll push the change later tonight.

File: pandas/tools/pivot.py (hunk 2 above)
Comment (jorisvandenbossche) [inline pandas/tools/pivot.py]: Ah, yes, that's good. But not yet the `pd.pivot` one, and that is apparently located here: https://github.com/pandas-dev/pandas/blob/master/pandas/core/reshape.py#L339

Comment (mroeschke) [inline pandas/tools/pivot.py]: Ah, thank you for bringing that to my attention. I'll push that change later tonight.

**Comment by mroeschke** [comment]
@jorisvandenbossche @jreback  Thanks for the feedback. Implemented the changes you guys suggested.

**Comment by jorisvandenbossche** [comment]
@mroeschke Thanks a lot! (feel free to do more PRs with doc improvements :-))

**Pull request merged**